components/provisioner/internal/apperrors/apperrors.go

```diff
@@ -1,10 +1,45 @@
 package apperrors
 
-import "fmt"
+import (
+	"fmt"
+)
+
+type ErrReason string
+type ErrComponent string
+
+const (
+	ErrDB                      ErrComponent = "db - provisioner"
+	ErrProvisionerK8SClient    ErrComponent = "k8s client - provisioner"
+	ErrProvisioner             ErrComponent = "provisioner"
+	ErrCompassDirectorClient   ErrComponent = "compass director client"
+	ErrCompassDirector         ErrComponent = "compass director"
+	ErrMpsOAuth2               ErrComponent = "mps oauth2"
+	ErrCompassConnectionClient ErrComponent = "compass connection client"
+	ErrCompassConnection       ErrComponent = "compass connection"
+	ErrGardenerClient          ErrComponent = "k8s client - gardener"
+	ErrGardener                ErrComponent = "gardener"
+	ErrClusterK8SClient        ErrComponent = "k8s client - cluster"
+	ErrKymaInstaller           ErrComponent = "kyma installer"
+)
+
+const (
+	ErrProvisionerInternal ErrReason = "err_provisioner_internal"
+
+	ErrDirectorNilResponse       ErrReason = "err_director_nil_response"
+	ErrDirectorRuntimeIDMismatch ErrReason = "err_director_runtime_id_mismatch"
+	ErrDirectorClientGraphqlizer ErrReason = "err_director_client_graphqlizer"
+
+	ErrCheckKymaInstallationState ErrReason = "err_check_kyma_installation_state"
+	ErrTriggerKymaInstallation    ErrReason = "err_trigger_kyma_installation"
+)
+
+type ErrCode int
+type CauseCode int
 
 const (
 	CodeBadGateway ErrCode = 502
 	CodeInternal   ErrCode = 500
+	CodeExternal   ErrCode = 501
```

Comment:
Hi, this External code is for the error from error component/category ErrMpsOAuth2("mps oauth2"), ErrKymaInstaller("kyma installer" hydroform) or ErrGardener("gardener").. It will be filtered out by KEB as https://github.com/kyma-project/control-plane/blob/66880df6700eb3fa10e67274b691b389af929d79/components/kyma-environment-broker/internal/provisioner/client.go#L189.

Do you think I could change it to InternalCode (I just wanted to differentiate this kind of error as the error from something external)?